frontend/packages/console-shared/src/components/formik-fields/BaseInputField.tsx

```diff
@@ -0,0 +1,50 @@
+import * as React from 'react';
+import { useField } from 'formik';
+import { FormGroup, ValidatedOptions } from '@patternfly/react-core';
+import { BaseInputFieldProps } from './field-types';
+import { getFieldId } from './field-utils';
+
+const BaseInputField: React.FC<BaseInputFieldProps & {
+  children: (props) => React.ReactNode;
+}> = ({
+  label,
+  helpText,
+  required,
+  children,
+  name,
+  onChange,
+  helpTextInvalid,
+  validated,
+  ...props
+}) => {
+  const [field, { touched, error }] = useField({ name, type: 'input' });
+  const fieldId = getFieldId(name, 'input');
+  const isValid = !(touched && error);
+  const errorMessage = !isValid ? error : '';
+  return (
+    <FormGroup
+      fieldId={fieldId}
+      label={label}
+      helperText={helpText}
+      helperTextInvalid={errorMessage || helpTextInvalid}
+      validated={!isValid ? ValidatedOptions.error : validated}
+      isRequired={required}
+    >
+      {children({
+        ...field,
+        ...props,
+        value: field.value || '',
+        id: fieldId,
+        label,
+        isValid,
+        'aria-describedby': `${fieldId}-helper`,
+        onChange: (value, event) => {
+          field.onChange(event);
+          onChange && onChange(event);
+        },
```

Comment:
```suggestion
      {children({
        ...field,
        ...props,
        value: field.value || '',
        id: fieldId,
        label,
        isValid,
        validated=!isValid ? ValidatedOptions.error : validated,
        'aria-describedby': `${fieldId}-helper`,
        onChange: (value, event) => {
          field.onChange(event);
          onChange && onChange(event);
        },
```
@debsmita1 you missed the validated prop on TextInput i.e children. which led to change in different css style and failure of integration test.
Integration test check for pf-m-success class on the text input after the validation of deploy image. and this class is added only when validated on TextInput is success(ValidatedOptions.success).

Reply:
Thank you for pointing this out